### PR TITLE
Upgrading unit tests to Junit 5 and upgrade to Java 17

### PR DESCRIPTION
**Pull request by jameswickham**
## Motivation

Upgrading unit tests for Junit 4 to 5

## Modification

Junit packages changed to the Junit 5 ones
Gradle build file Junit dependency version increased to 5.
Small tweak to assertThrows method as the ordering of the parameters has changed in Junit 5(where it expects your error message).
Also updating build files to use Java 17 and the required version of Gradle

## PR Checklist

- [x] been self-reviewed.
- [n/a] Added javadocs for most classes and all non-trivial methods
- [n/a] Added supporting annotations (like @XStreamAlias / @ComponentProfile)
- [n/a] Added DefaultValue annotation when there is a default value (e.g. @DefaultValue('true'))
- [n/a] Added validation annotation (@NotNull...) when required and add @Valid when nested class has some validation
- [n/a] Checked that @NotNull and @NotBlank annotations have a meaningful message when appropriate.
- [n/a] Checked that new 3rd party dependencies are appropriately licensed
- [n/a] Added comments explaining the "why" and the intent of the code wherever it would not be obvious for an unfamiliar reader
- [x] Added unit tests or modified existing tests to cover new code paths
- [n/a] Tested new/updated components in the UI and at runtime in an Interlok instance
- [n/a] Reviewed java class members so that missing annotations are added (InputFieldDefault/ComponentProfile etc)
- [n/a] Checked that javadoc generation doesn't report errors
- [n/a] Checked the display of the component in the UI
- [n/a] Remove any config/license annotations
- [n/a] Check the gradle build file to make sure the dependencies section is more explicit "implementation/api".

## Result

What's the end result for the user
No changes

## Testing

How can I test this if I'm reviewing this.
If the gradle build is successful and code coverage hasn't dropped then you know it's worked as it means all unit tests ran successfully.

### CODE REVIEW

**Comment by jameswickham** [comment]
@sebastien-belin-adp @aaron-mcgrath-adp @higgyfella 
I have spotted that while the unit tests all run successfully the change to Junit 5 has caused a big drop in code coverage

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.1-bin.zip
+networkTimeout=10000
```
Comment (sebastien-belin-adp) [inline gradle/wrapper/gradle-wrapper.properties:4]: I don't think we need `networkTimeout=10000`

File: build.gradle [hunk 2]
```diff
@@ -28,6 +28,9 @@ ext {
   organizationUrl = "http://interlok.adaptris.net"
   slf4jVersion = '2.0.3'
   mockitoVersion = '4.9.0'
+  
+  interlokCommonsOfflineJavadocDir = project.rootDir.getCanonicalPath() + "/interlok-common/build/docs/javadoc"
```
Comment (sebastien-belin-adp) [inline build.gradle:32]: I don't think we need this line and maybe not the one below either

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (jameswickham) [reply gradle/wrapper/gradle-wrapper.properties]: Must confess I saw it was in the properties file for interlok-core so I copied it over

File: build.gradle (hunk 2 above)
Comment (jameswickham) [reply build.gradle]: same as the comment above, saw it in interlok-core and thought it was one of the required changes to the build file for junit 5 and java 17

File: .circleci/config.yml
[the remark this answers is not in the document]
Comment (jameswickham) [reply .circleci/config.yml:1]: removed all the java8 and 11 stuff and now it just has a build job for java 17. See conversation about this in interlok-json PR for more: https://github.com/adaptris/interlok-json/pull/435

**Review by sebastien-belin-adp — APPROVED** [review]

**Review by sebastien-belin-adp — APPROVED** [review]

**Pull request merged**